### PR TITLE
Introduce 2FA functionality for non end-user roles on the site

### PR DESCRIPTION
**Pull request by johangant**
This aims to provide Two Factor Authentication (2FA) support for administrative users to better secure the authentication process. This is not expected to be mandatory for end-users.

At present, this PR assumes an encryption key is defined and stored using lockr.io - this may need to change as Wunder devs have advised that this can also be stored in the Ansible vault, but shouldn't be a blocker for general review.

### CODE REVIEW

File: drupal/composer.lock [hunk 1]
```diff
@@ -3550,17 +3782,17 @@
         },
         {
             "name": "drupal/entity_reference_revisions",
-            "version": "1.5.0",
+            "version": "1.3.0",
```
Comment (kirkkala) [inline drupal/composer.lock]: @johangant watch out when adding something to composer to not revert updates already merged to master. See also `drupal/paragraphs` which should remain at 1.3.0 as per [this commit](https://github.com/wunderio/client-UK-FSA-beta/commit/e9e22a1c4b5b35679b505dede7ba1b4253f70d67) in master.

File: drupal/sync/config_ignore.settings.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@ ignored_config_entities:
   0: system.site
   2: config.fsa_ratings
   4: config.fsa_consultations
-  6: fsa_content_reminder.settings
```
Comment (kirkkala) [inline drupal/sync/config_ignore.settings.yml]: Maybe an accidental removal upon conflict resolving? Please retain `fsa_content_reminder.settings` in ignore file unless a very good reason to remove :)

Comment (johangant) [reply drupal/sync/config_ignore.settings.yml]: Definitely accidental, thanks for spotting!

File: drupal/composer.lock (hunk 1 above)
Comment (johangant) [reply drupal/composer.lock]: Noted, thanks for spotting 👍

**Comment by johangant** [comment]
@kirkkala Hard to say exactly why composer.lock changed to change version numbers for paragraphs and entity_reference_revisions packages, it wasn't expected in this feature. Rather than revert changes to the lock file which could easily be lost I've specified the versions of each in the `composer.json` file. That might make future updates slightly more effort; `composer update` will always fetch the same version for those two.

I can't think of a safer and more convenient option to lock those package versions down at this stage and it sounds like they're important for ongoing translation features work.

**Comment by kirkkala** [comment]
@johangant sure, would work to restrict the version in composer.json but like you said more effort on future updates. For me (so far) it has been working fine on composer.lock conflicts to 

1. Remove drupal/composer.lock
2. run `composer update nothing` inside box
3. Check diff on `drupal/composer.lock` it updated what it really should have
4. Profit

Let me know if that works and then maybe revert the composer.json changes?

**Comment by kirkkala** [comment]
@johangant yes, both `paragraphs` and `entity_reference_revisions` should be at latest versions (paragraphs in 8.x-1.3 and entity_reference_revisions in 8.x-1.5)

**Pull request closed without merging**